### DISCUSSION TITLE
How to give names to duplicatable fields?

### DISCUSSION THREAD

**Discussion by morceaudebois** [directus/directus#9758] [Archive]
Hi! First, I'm new to Directus and wanted to say how amazed I was at how clean, fast and powerful it looks. I really hope I'm gonna be able to replace my WordPress + ACF installation with it!

There's something I don't really understand how to do though. I have a data model with a repeater field I made duplicatable (via the "Duplication" zone at the bottom of the page). When I go to my model on the Content collection page, I can add and remove the repeaters just like I want, which is great. However, I can't name them properly. The displayed title is always some raw data, which isn't practical at all.

I went into the options for my repeater field, and found the Display panel which seems related to it, but I can't seem to find a way to put a simple, editable title for each duplicated repeater field. The [documentation about that panel](https://docs.directus.io/configuration/data-model/#display) is.. scarce, to say the least.

I made a small video for context.

Thanks a lot for your help :)

https://user-images.githubusercontent.com/50768582/141478698-ccfc1d30-a733-4b76-b9d3-0a7933c83456.mp4

**Comment by benhaynes** [comment]
Thanks!!

I'm not sure which Display will best show the values of a repeater (probably JSON), but raw is expected to show what you're seeing there. We might need a new Display for showing this in a nice way.

**Reply by morceaudebois** [reply]
Yeah, that makes sense :) I tried the other two options of course, I think there would be something to be done with the formatted JSON value one, but it would require a single title field to be added for each duplicate and it doesn't seem to be possible.

I really need this feature for my project, how hard would it be to add it myself?

**Comment by rijkvanzanten** [comment]
For what it's worth, the "Duplication" setting of the collection setup page controls what fields will be duplicated when you use the "Save as Copy" option of the app, it does not control anything in the "repeatability" of the repeater interface.

As for the display, what output are you looking for exactly? The current "formatted json" display allows you to render a count on the row, with a popover detailing the exact items, as follows:

<img width="493" alt="CleanShot 2021-11-12 at 10 50 43@2x" src="https://user-images.githubusercontent.com/9141017/141495101-1e3fe9bf-69d0-470b-9617-55502386d308.png">

**Reply by morceaudebois** [reply]
Oh I see, I didn't realise. Thanks!

I'd like it to be editable by the user. It would be to make a restaurant menu, each duplicated repeater would be used for a category in the menu, a repeater for "Wine" and another one for "Desserts" for example.
